src/plugins/workspace/public/plugin.ts

```diff
@@ -78,18 +84,27 @@ export class WorkspacesPlugin implements Plugin<{}, {}> {
     return {};
   }
 
-  private async _changeSavedObjectCurrentWorkspace() {
-    const startServices = await this.core?.getStartServices();
-    if (startServices) {
-      const coreStart = startServices[0];
-      coreStart.workspaces.client.currentWorkspaceId$.subscribe((currentWorkspaceId) => {
-        coreStart.savedObjects.client.setCurrentWorkspace(currentWorkspaceId);
+  private _changeSavedObjectCurrentWorkspace() {
+    if (this.coreStart) {
+      this.coreStart.workspaces.client.currentWorkspaceId$.subscribe((currentWorkspaceId) => {
+        this.coreStart?.savedObjects.client.setCurrentWorkspace(currentWorkspaceId);
       });
```

Comment:
should we unsubscribe this in `stop` lifecycle?

Reply:
Good point, looks like we should, updated now